### DISCUSSION TITLE
Enforce HTTPS: SSL Redirect

### DISCUSSION THREAD

**Discussion by ottovw** [vercel/next.js#21584] [Ideas]
**Describe the feature you'd like to request**

- We need an option to redirect client requests to use HTTPS
- Should support Header X-Forwared-To if client is behind reverse proxy (Nginx, Heroku, ...)

**Describe the solution you'd like**

- An config flag to enable HTTPS redirect
- Should support Header X-Forwared-To if client is behind reverse proxy (Nginx, Heroku, ...)

**Describe alternatives you've considered**

Custom server works. We beliebe, this should be a default feature.

**Comment by RenanGalvao** [comment]
Is there some workaround? I assume this could be done with getServerSideProps but my app has only one page and I already used ISR on it. I've setted up with create next-app.

**Comment by ThibautJ** [comment]
You say that custom server works. But a custom server usage implies no Automatic Static Optimization for example.
So, that's not working in that particular case.
Have you other workaround to enforce https without using custom server?

**Reply by EloB** [reply]
Is there any progress on this one? Seems kind of important especially if you want to get good score on "Lighthouse".
